src/txn/txn_rollback_to_stable.c

```diff
@@ -304,15 +282,30 @@ __rollback_row_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_PAGE *page, WT_ROW
 #endif
 
     /* Allocate buffers for the data store and history store key. */
-    WT_RET(__wt_scr_alloc(session, 0, &key));
     WT_ERR(__wt_scr_alloc(session, 0, &hs_key));
     WT_ERR(__wt_scr_alloc(session, 0, &hs_value));
 
-    WT_ERR(__wt_row_leaf_key(session, page, rip, key, false));
+    /* Unpack a row cell. */
+    if (rip != NULL) {
+        WT_ERR(__wt_scr_alloc(session, 0, &key));
```

Comment:
Assert that `cip` NULL whenever `rip` is not NULL and vice-versa.

Reply:
Fixed.